docs/route.md

```diff
@@ -125,94 +125,46 @@ This is a simple property, so you can both read it and write to it.
 ```
 
-You can pass other attributes, too, and you can also specify the tag name used.
-
-```javascript
-m(m.route.Link, {
-	// Any hyperscript selector is valid here - it's literally passed as the
-	// first parameter to `m`.
-	selector: "span",
-	options: {replace: true},
-	params: {key: "value"},
-	href: "/test",
-	disabled: false,
-	class: "nav-link",
-	"data-foo": 1,
-	// and other attributes
-}, "link name")
-```
-
-Magic attributes used by this selector (except `href` and `disabled`) *are* removed while proxying, so you won't have an odd `selector="span"` or `options="[object Object]"` attribute show up in your link's DOM node. The above vnode renders to this hyperscript, assuming the prefix is the default `#!`:
-
-```javascript
-m("span", {
-	href: "#!/test",
-	onclick: function(e) {
-		// ...
-	},
-	disabled: false, // Only if you specify it
-	class: "nav-link",
-	"data-foo": 1,
-	// and other attributes
-})
-```
+Links accept a selection of special attributes: 
+* `selector` is what would be passed as the first argument to [`m`](hyperscript.md): any selector is valid, including non-`a` elements.
+* `params` & `options` are the arguments with the same names as defined in [`m.route.set`](#mrouteset).
+* `disabled`, if true, disables routing behaviour and any bound `onclick` handler, and attaches a `data-disabled="true"` attribute for accessibility hints; if the element is an `a`, the `href` is removed.
 
-You can also prevent navigation by, in an `onclick` handler, invoking `e.preventDefault()` or returning `false`. This is the same way you block other events, so it's pretty natural.
+*Routing behaviour cannot be prevented using the event handling API: use `disabled` instead.*
 
 ```javascript
 m(m.route.Link, {
-	href: "/test",
-	onclick: function(e) {
-		// Do things...
-		if (notReady()) e.preventDefault()
-	}
+	href: "/foo",
+	selector: "button.large",
+	disabled: true,
+	params: {key: "value"},
+	options: {replace: true},
 }, "link name")
-```
-
-This supports full accessibility for both `a` and `button`, via a `disabled` attribute. This ensures [no `href` attribute or `onclick` handler is set](https://css-tricks.com/how-to-disable-links/) and that an `"aria-disabled": "true"` attribute *is* set. If you are passing an `onclick` handler already, that's dropped. (You can work around this by adding it directly in a [lifecycle hook](lifecycle-methods.md).) The `disabled` attribute is itself proxied to the element or component, so you can disable routed `<button>`s and the like.
 
-```javascript
-// This does the right thing and the accessible thing for you.
-m(m.route.Link, {disabled: disabled, href: "/test"}, "disabled")
-
-// It renders to this hyperscript, assuming the prefix is the default one:
-m("a", {
-	href: "#!/test",
-	disabled: disabled,
-	"aria-disabled": disabled ? "true" : false,
-	onclick: disabled ? null : function(e) {
-		// ...
-	},
-})
+// Renders to:
+// <button disabled class"large">link name</a>
```

Comment:
I know the scope of the PR is just to address the documentation being clearer, and it definitely is, but I find it really surprising you can use m.route.Link with non-anchors and that mithril accounts for that.

Reply:
I plan to redress this in a later PR but it requires backing out this feature altogether. In the meantime I realise I forgot to put the ARIA attribute in.